tests/testthat/test-shinytest2_filteredData_DF.R

```diff
@@ -0,0 +1,231 @@
+  it("'Active Filter Summary'", {
+    app_driver <- local_app_driver()
+
+    selector_collapsable <- get_class(".accordion-item:nth-of-type(1) > div.accordion-collapse")
+    out <- app_driver$get_js(selector_collapsable)
+    testthat::expect_length(out, 4L)
+    testthat::expect_true("show" %in% unlist(out[[1L]]))
+
+    selector <- ".filter-panel > .teal-slice:nth-of-type(1) .accordion-button"
+    app_driver$click(selector = selector)
+    app_driver$wait_for_idle()
+
+    out <- app_driver$get_js(selector_collapsable)
+    testthat::expect_length(out, 4L)
+    testthat::expect_false("show" %in% unlist(out[[1L]]))
+    testthat::expect_equal(
+      app_driver$get_text(
+        paste(
+          "#filter_panel-overview-main_filter_accordion > div >",
+          "div.accordion-header > button > div.accordion-title"
+        )
+      ),
+      "Active Filter Summary"
+    )
+
+    table <- "#filter_panel-overview-table > table"
+    text <- app_driver$get_text(table)
+
+    testthat::expect_equal(
+      clean_text(text),
+      c("Data Name", "Obs", "iris", "50/150", "mtcars", "5/32")
+    )
+  })
+
+  it("'Filter Data'", {
+    app_driver <- local_app_driver()
+
+    selector_collapsable <- get_class(".accordion-item:nth-of-type(1) > div.accordion-collapse")
+    out <- app_driver$get_js(selector_collapsable)
+    testthat::expect_length(out, 4L)
+    testthat::expect_true("show" %in% unlist(out[[2L]]))
+
+    selector <- ".filter-panel > #filter_panel-active.teal-slice > div > div > div > button"
+    app_driver$click(selector = selector)
+    app_driver$wait_for_idle()
+
+    out <- app_driver$get_js(selector_collapsable)
+    testthat::expect_length(out, 4L)
+    testthat::expect_false("show" %in% unlist(out[[2L]]))
+
+
+    testthat::expect_equal(
+      app_driver$get_text(paste0(
+        "#filter_panel-active-main_filter_accordion > div > ",
+        "div.accordion-header > button > div.accordion-title"
+      )),
+      "Filter Data"
+    )
+  })
+
+  it("filters for a dataset", {
+    app_driver <- local_app_driver()
+
+    selector_collapsable <- get_class(".accordion-item:nth-of-type(1) > div.accordion-collapse")
+    out <- app_driver$get_js(selector_collapsable)
+    testthat::expect_length(out, 4L)
+    testthat::expect_true("show" %in% unlist(out[[3L]]))
+
+    selector <- "#iris * button"
+    app_driver$click(selector = selector)
+    app_driver$wait_for_idle()
+
+    out <- app_driver$get_js(selector_collapsable)
+    testthat::expect_length(out, 4L)
+    testthat::expect_false("show" %in% unlist(out[[3L]]))
+
+    # The text is still in DOM
+    text <- app_driver$get_text("#filter_panel-active-iris-filter-iris_Species-inputs-selection")
+    testthat::expect_equal(
+      clean_text(text),
+      c("setosa (50/50)", "versicolor (50/50)", "virginica (50/50)")
+    )
+  })
+})
+
+testthat::describe("Clicking remove buttons removes the visibility of", {
```

Comment:
It is not visibility. Element is completely removed
```suggestion
testthat::describe("Clicking remove buttons removes the html element of associated filter-state-card", {
```